src/main/java/org/diego/tutorial/car/resources/CarResource.java

```diff
@@ -74,11 +109,27 @@ public Response getCar(@PathParam("id") long id) {
 						.build();
 	}
 	
+	/**
+	 * Method that updates the information related to an specific car
+	 * @param id Identifier of the car that should be updated
+	 * @param car Car object with the new information
+	 * @return Car updated
+	 */
 	@PUT
 	@Path("/{id}")
 	public Response updateCar(@PathParam("id") long id,
 							Car car) {
 		car.setId(id);
+		
+		List<String> validationErrors = CarValidator.validateAddAndUpdate(car);
+		if (!validationErrors.isEmpty()) {
+			String message = "Request to update car with non valid fields: <p>";
+			for (String validationError : validationErrors) {
+				message += validationError + "<p>";
+			}
+			throw new BadRequestException(message);
+		}
+		

```

Comment:
this should be encapsulated into some method somewhere else

Reply:
Updated with encapsulation methods.